shared-types.d.ts

```diff
@@ -0,0 +1,220 @@
+// TypeScript Version: 3.0
+
+/**
+ * A location in a string or buffer
+ */
+export interface Point {
+  line: number
+  column: number
+  offset: number
+  _index?: number
+  _bufferIndex?: number
```

Comment:
For the underscore props, as those are kinda for internal use in this project, should they be left undocumented?

Is there some comment to add here to say to TS that it shouldn’t be exported to library users?

Reply:
We could tag these with tsdoc
```
/**
 * @internal
 */
```
I'm cautious about removing them entirely, since extensions/plugins may leverage some of the internal attributes.